ssl/statem/extensions.c

```diff
@@ -911,8 +911,6 @@ static int final_server_name(SSL *s, unsigned int context, int sent,
 
     case SSL_TLSEXT_ERR_NOACK:
         s->servername_done = 0;
-        if (s->server && s->session->ext.hostname != NULL)
-            s->ext.early_data_ok = 0;
         return 1;
```

Comment:
Do we have another check elsewhere that would enforce the edsess not having a hostname set?

Reply:
The hostname stored in the session is the SNI sent in the ClientHello that established that session. It does *not* indicate what the server's response was, i.e. whether it was acknowledged or not (actually we don't store that anywhere). So the mere presence of a hostname in the session that we are not acknowledging this time around does not (necessarily) indicate an SNI inconsistency. There is actually a corner case here where the client sends a hostname and the server acknowledges it to establish the session, and then the client resumes sending through the same SNI hostname but the server doesn't acknowledge it. Technically this is an SNI inconsistency. We can't actually detect that inconsistency at the moment because we don't store the acknowledgement status, so we're making the simplifying assumption that if we didn't acknowledge this time, then we didn't last time either. The only other solution is to add a field to the session to store the acknowledgement status. I wasn't sure it was worth it.

Reply:
Ah, got it.  I didn't check whether we had to accept the SNI in order to put it in the session.
(I don't think it's worth it to add a field just for this.)